### PR TITLE
dmp-3626: ARM Pull response log error in DB

### PR DESCRIPTION
**Pull request by hemantasharma1129**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
DMP-3626

### Change description ###
- Log ARM error description in DB.
- Update Ingestion date in EOD after successful processing.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java [hunk 1]
```diff
@@ -1192,6 +1194,154 @@ void batchProcessResponseFiles_WithCaseDocumentInvalidResponseFilenames() throws
...
+        String manifestFile1 = "DARTS_" + manifest1Uuid + ".a360";
+
+        ExternalObjectDirectoryEntity armEod = dartsDatabase.getExternalObjectDirectoryStub().createExternalObjectDirectory(
+            caseDocument,
+            dartsDatabase.getObjectRecordStatusEntity(ARM_DROP_ZONE),
+            dartsDatabase.getExternalLocationTypeEntity(ARM),
+            UUID.randomUUID()
+        );
+        armEod.setTransferAttempts(1);
+        armEod.setTransferAttempts(1);
```
Comment (karen-hedges) [inline src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java:1294]: You have duplicated this line 1293, 1294

File: src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java [hunk 2]
```diff
@@ -1192,6 +1194,154 @@ void batchProcessResponseFiles_WithCaseDocumentInvalidResponseFilenames() throws
...
+        when(armDataManagementConfiguration.getTempBlobWorkspace()).thenReturn(fileLocation);
+        when(armDataManagementConfiguration.getContinuationTokenDuration()).thenReturn("PT1M");
+        when(armDataManagementConfiguration.getManifestFilePrefix()).thenReturn(PREFIX);
+        when(armDataManagementConfiguration.getFileExtension()).thenReturn("a360");
+
+        // when
+        armBatchProcessResponseFiles.processResponseFiles();
+
+        // then
+        ExternalObjectDirectoryEntity foundAnnotationEod = dartsDatabase.getExternalObjectDirectoryRepository()
```
Comment (karen-hedges) [inline src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java:1338]: This should probably be foundCaseDocumentEod not annotation?

**Review by karen-hedges — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java (hunk 1 above)
Comment (hemantasharma1129) [reply src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java]: Removed.

File: src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java (hunk 2 above)
Comment (hemantasharma1129) [reply src/integrationTest/java/uk/gov/hmcts/darts/arm/service/ArmBatchProcessResponseFilesIntTest.java]: Updated.

**Pull request merged**